### PR TITLE
Add API key header to HTTP requests

### PR DESCRIPTION
**Pull request by aphexyuri**

### CODE REVIEW

**Review by alecananian — APPROVED** [review]

File: Assets/Treasure/TDK/Runtime/Infrastructure/TDKTimeKeeper.cs [hunk 1]
```diff
@@ -29,6 +29,9 @@ private IEnumerator GetServerTime()
 
             using (UnityWebRequest webRequest = UnityWebRequest.Get(endpoint))
             {
+                // Set the API key header
+                webRequest.SetRequestHeader("x-api-key", TDK.Instance.AppConfig.ApiKey);
```
Comment (lefarchi) [inline Assets/Treasure/TDK/Runtime/Infrastructure/TDKTimeKeeper.cs:33]: Is it needed for the time keeper? Even if its a public endpoint

**Review by lefarchi — COMMENTED** [review]
Do we need to add this header in the `Get` and `Post` methods in `TDK.API.cs` or is it not necessary

**Review by lefarchi — APPROVED** [review]
Reading again, I see now that its only for darkmatter ✅

File: Assets/Treasure/TDK/Runtime/Infrastructure/TDKTimeKeeper.cs (hunk 1 above)
Comment (RandolfTjandra) [reply Assets/Treasure/TDK/Runtime/Infrastructure/TDKTimeKeeper.cs:33]: not for the time keeper. just the /ingress apis

**Review by RandolfTjandra — APPROVED** [review]

**Pull request merged**